src/lib.rs

```diff
@@ -101,7 +97,13 @@ macro_rules! expect_two {
 /// // Claims is a struct that implements Deserialize
 /// let token_data = decode::<Claims>(&token, "secret", Algorithm::HS256, &Validation::default());
 /// ```
-pub fn decode<T: Deserialize>(token: &str, key: &[u8], algorithm: Algorithm, validation: &Validation) -> Result<TokenData<T>> {
+pub fn decode<T>(token: &str,
+                 key: &[u8],
+                 algorithm: Algorithm,
+                 validation: &Validation)
+                 -> Result<TokenData<T>>
+    where for<'a> T: Deserialize<'a>
```

Comment:
This would be more readable as `T: DeserializeOwned`. This function requires that `T` can deserialize without borrowing any data because the string it is deserialized from is thrown away before this function returns.